frontend/catalyst/jax_extras/tracing.py

```diff
@@ -399,9 +556,423 @@ def make_jaxpr_f(*args, **kwargs):
+    return _input_type_to_tracers(maker, in_aval)
+
+
+def output_type_to_tracers(
+    out_type: OutputType,
+    out_consts: List[DynamicJaxprTracer],
+    in_tracers: List[DynamicJaxprTracer],
+    maker: Callable[[AbstractValue], DynamicJaxprTracer],
+) -> List[DynamicJaxprTracer]:
+    """Creates an expanded list of tracers representing an output values of a Jaxpr program
+    based on the ``out_type`` of this program. The resulting tracers might be nested as required by
+    the Jax dynamic API and might contain ``in_tracers`` of the same Jaxpr program."""
+    out_tracers = []
+    for aval, _ in out_type:
+        if isinstance(aval, DShapedArray):
+            shape = [
+                (
+                    [*out_consts, *in_tracers][d.val]
+                    if isinstance(d, InDBIdx)
+                    else out_tracers[d.val] if isinstance(d, OutDBIdx) else d
+                )
+                for d in aval.shape
+            ]
+            aval = aval.update(shape=tuple(shape))
+        out_tracers.append(maker(aval))
+    return out_tracers
+
+
+TracerLike = TypeVar("TracerLike")
+
+
+def infer_input_type_unshared(inputs: List[TracerLike]) -> InputType:
+    """Infer the input type of a function having `inputs` Jax tracers."""
+
+    def _is_tracer_like(x):
+        return hasattr(x, "aval")
+
+    expl_ins = inputs
+    impl_avals = []
+    expl_avals = []
+    for o in expl_ins:
+        assert _is_tracer_like(o), (o,)
+        if isinstance(o.aval, DShapedArray):
+            shape2 = []
+            for d in o.aval.shape:
+                if _is_tracer_like(d):
+                    shape2.append(DBIdx(len(impl_avals)))
+                    impl_avals.append(d.aval)
+                else:
+                    shape2.append(d)
+            expl_avals.append(o.aval.update(shape=tuple(shape2)))
+        else:
+            expl_avals.append(o.aval)
+    return (*[(i, False) for i in impl_avals], *[(i, True) for i in expl_avals])
+
+
+@dataclass
+class ExpansionStrategy:
+    """Describe the settings affecting the Jax dyanmic API tracing of the nested programs.
+    Args:
+        axes_specs: Axes specification used to convert static dimensions to the dynamic ones
+        input_unshare_variables: Treat each dynamic dimension as a distinct dimension, even if some
+                                 of them are described by the same dimension variables.
+        output_include_indbidx_vars: Include variables corresponding to the InDBIdx references in
+                                     the result
+        output_force_arg0_outdbidx: Force all references pointing to the first arguments to be
+                                    OutDBIdx. This is typically required to avoid input references
+                                    to the loop iteration variable.
+    """
+
+    axes_specs: Sequence[AbstractedAxesSpec] | None
+    input_unshare_variables: bool
+    output_include_indbidx_vars: bool
+    output_force_arg0_outdbidx: bool
```

Comment:
```suggestion
    output_include_in_db_idx_vars: bool
    output_force_arg0_out_db_idx: bool
```

It is hard to understand what these refer without spaces.

Reply:
I would leave these without changes because InDBIdx and OutDBIdx are part of Jax API

Reply:
I know, but I was unable to parse `indbidx` I thought it could mean Index Bid X for example :sweat_smile: This is just a variable name change to make it more legible. I think Changing them to camelCase could also work, but since everything is snake case I proposed the above. If you really insist, ok.

Reply:
Well, my argument is - leaving it without separator would make the search easier. It is really a kind of standard (but ye - weird) name, so lets keep it for them.